package.json

```diff
@@ -32,11 +32,15 @@
     "start": "start-storybook -p 9001 -c storybook -s ./storybook/static",
     "build": "lerna run --parallel build -- -- --env production",
     "build:storybook": "build-storybook -c storybook -o ./storybook/public -s ./storybook/static",
+    "setup": "npm run setup:hooks && npm run init",
```

Comment:
I don't want to have multiple 'setup' or 'init' scripts. It gets confusing knowing which one to run when you're setting up the repo locally.

Additionally there's nothing that requires a user to install a hook. I believe when you `npm install` and husky is there it installs the hooks for you by default, so there's no avoiding the hooks.

On that note, I think having us run all of lint and all of tests before pushing is a little heavy. It's going to make for an annoying dev experience when everytime I push I have to wait for tests to run. As the codebase gets bigger that'll become slower and slower. Also should be something that is handled by the CI (coming soon!).

However I do think we should have a precommit hook when running publish. We should run `npm run build` before publish to make sure we publish the right assets.

Reply:
Fair enough. Given how init is used, I could almost see renaming it to `npm run install`, though that could get confusing with `npm install`. Maybe `npm run bootstrap`? As for the hooks setup script, do you believe that installing them should be entirely standalone then? Thoughts?

Git hooks do need to be installed in some fashion since the `.git/hooks/` directory that is used by default doesn't get committed. If we use husky, yes that would be automated. If you strongly think that we should add it, then we can. It just feels bad since this current implementation is 2-3 lines of code without a library.

Sure we could totally omit the tests when we have a CI process in place. Right now, I think the tests are fine since they currently don't take very long, but we can remove as soon as that's automated.

As for publish, yeah we should flesh out the script to do so more to add protections and automation around building. Made a ticket to do that, since it's outside the scope of this PR: https://jira.mongodb.org/browse/PD-239.